backend/mutations.py

```diff
@@ -231,6 +234,66 @@ def mutate(self, info, id):
         return DeleteGroupMutation(success=success)
 
 
+class JoinGroupMutation(graphene.Mutation):
+    """ Allows a specific user to join a specific group """
+    success = graphene.Boolean()
+
+    class Arguments:
+        user_id = graphene.String()
+        group_id = graphene.String()
+
+    def mutate(self, info, user_id, group_id):
+
+        group = UpdateGroupMutation.getGroup(group_id)
+        user = UpdateUserMutation.getUser(user_id)
+        expected_sizes = (len(group.members) + 1, len(user.groups) + 1)
+        group.members.append(user)
+        user.groups.append(group)
+        actual_sizes = (len(group.members), len(user.groups))
+
+        if expected_sizes == actual_sizes:
+            user.save()
+            group.save()
+            success = True
+        else:
+            success = False
```

Comment:
Good validation check! I could see this being a much larger problem if we don't keep it in sync. One edge case though would be this system failing halfway through adding a member to the list  and then no sign ups could possibly happen in the future because one of the lists will always be ahead of the other. Just something to consider.